Added occ install option for database-port

Faldon opened the pull request:
# Implemented non-standard database port on occ install
- Added optional input option --database-port in maintenance:install command
- Refactored the databases to store the database port as string, defaulting to an empty string
- Kept parsing hostname for port option in psql and mysql when no port was provided
- Changed the PostgreSQL connection error message
### Tested
- with MySQL TCP port
- with MySQL Unix Socket
- with PostgreSQL
### ToDo
- Testing with Oracle

Solves #271

Commit:
Added occ install option for database-port.
Extended the database setup to store the database port.
Changed the PostgreSQL connection error message for clarification.

nickvergessen commented on core/Command/Maintenance/Install.php:
```diff
@@ -50,6 +50,7 @@ protected function configure() {
 			->addOption('database', null, InputOption::VALUE_REQUIRED, 'Supported database type', 'sqlite')
 			->addOption('database-name', null, InputOption::VALUE_REQUIRED, 'Name of the database')
 			->addOption('database-host', null, InputOption::VALUE_REQUIRED, 'Hostname of the database', 'localhost')
+			->addOption('database-port', null, InputOption::VALUE_OPTIONAL, 'Port the database is listening on')
```
Should be `InputOption::VALUE_REQUIRED`

Options are all optional, but when you specify it, it needs a value.

nickvergessen commented on lib/private/Setup/OCI.php:
```diff
@@ -63,12 +63,14 @@ public function validate($config) {
 	public function setupDatabase($username) {
 		$e_host = addslashes($this->dbHost);
+		// adding slashes for security reasons
+		$e_port = addslashes($this->dbPort);
```
cast to int instead?

nickvergessen commented on lib/private/Setup/PostgreSQL.php:
```diff
@@ -34,8 +34,11 @@ public function setupDatabase($username) {
+		// adding port support through installer
+		if(!empty($this->dbPort)) {
+			// adding slashes for security reasons
+			$port = addslashes($this->dbPort);
```
same here

Faldon commented:
Ofc, casting to int is much more appropriate. And in case the port was provided with the hostname, addslashes is already executed on the string.

Commit:
Changed the input option for database-port to required when parameter was provided.
Added casting database port to int for input sanitation in pgsql and oci connections.

nickvergessen commented:
:+1:

MorrisJobke commented:
Tested and works 👍
